scripts/CMakeLists.txt

```diff
@@ -30,14 +30,18 @@ ie_shellcheck_process(DIRECTORY "${OpenVINO_SOURCE_DIR}"
 ie_cpack_add_component(setupvars REQUIRED)
 
 if(UNIX)
-    install(PROGRAMS setupvars/setupvars.sh
-            DESTINATION .
-            COMPONENT setupvars)
+    set(_setupvars_file setupvars/setupvars.sh)
 elseif(WIN32)
-    install(PROGRAMS setupvars/setupvars.bat
-            DESTINATION .
-            COMPONENT setupvars)
+    set(_setupvars_file setupvars/setupvars.bat)
+    # Update setupvars.bat to support non "Release" build types
+    file(READ "${_setupvars_file}" _setupvars_content)
+    string(REPLACE "Release" ${CMAKE_BUILD_TYPE} _setupvars_content ${_setupvars_content})
```

Comment:
`CMAKE_BUILD_TYPE` can be wrong at the configuration time when using multi-config generator (e.g. Visual Studio). It can only be deduced at the build time. It looks like current setupvars.bat just does not support required functionality.

I see two ways to solve this:
* add cfg parameter to `setupvars.bat`, e.g. `setupvars.bat -cfg Debug`. Then it will be possible to have multiple kinds of binaries and switch between them when needed. Script can verify that folder with cfg name exists for convenience.
* limit allowed configurations to two: _Release_ and _Debug_ (as written in the setupvars), provide options to enable debug info in release builds (to produce RelWithDebInfo) or size-related optimizations (MinSizeRel). This approach is implemented in OpenCV. It may look not optimal, but works pretty well for many users. Advanced users can be provided with ways of adding extra configurations.

Reply:
Good catch. Have updated the PR to handle this case.

I want to limit this PR to support [USE_BUILD_TYPE_SUBFOLDER](https://github.com/openvinotoolkit/openvino/blob/master/cmake/developer_package/IEDevScriptsConfig.cmake#L116) logic correctly, not implement full multi config support for setupvars.

This aim is to enable address sanitizer on RelWithDebInfo config set explicitly.